frame/support/src/traits/dispatch.rs

```diff
@@ -38,6 +38,55 @@ pub trait EnsureOrigin<OuterOrigin> {
 	fn successful_origin() -> OuterOrigin;
 }
 
+/// The "OR gate" implementation of `EnsureOrigin`.
+///
+/// Origin check will pass if `L` or `R` origin check passes. `L` is tested first.
+pub struct EnsureOneOf<L, R>(sp_std::marker::PhantomData<(L, R)>);
```

Comment:
I liked this refactor which gets rid of unnecessary generics. maybe we can isolate in another PR if this one doesn't get merged

Reply:
Yes, sure. Will move this code into the another PR.